wgsl/index.bs

```diff
@@ -4180,6 +4180,242 @@ TODO: deduplicate these tables
+textureSampleLevel(texture : texture_depth_cube,               sampler : sampler, coords : vec3<f32>,                    level : f32                    ) -> f32
+textureSampleLevel(texture : texture_depth_cube_array,         sampler : sampler, coords : vec3<f32>, array_index : u32, level : f32                    ) -> f32
+```
+
+**Parameters:**
+
+<table class='data'>
+  <tr><td>`texture`<td>
+  The [sampled](#sampled-texture-type) or [depth](#texture-depth) texture to
+  sample.
+  <tr><td>`sampler`<td>
+  The [sampler type](#sampler-type).
+  <tr><td>`coords`<td>
+  The texture coordinates used for sampling.
+  <tr><td>`array_index`<td>
+  The 0-based texture array index to sample.
+  <tr><td>`level`<td>
+  The mip level to sample from, with 0 representing the largest level.
+  Fractional values may interpolate between two levels.
+  <tr><td>`offset`<td>
+  The optional texel offset applied to the unnormalized texture coordinate
+  before sampling the texture. This offset is applied before applying any
+  texture wrapping modes. `offset` values may range from `[-8..7]` in all
+  dimensions. Values outside of this range will be clamped to the nearest limit.
+</table>
+
+**Returns:**
+
+The sampled value.
+
+
+### `textureSampleGrad` ### {#texturesamplegrad}
+
+Samples a texture using explicit gradients.
+
+```rust
+textureSampleGrad(texture : texture_sampled_2d<type>,         sampler : sampler, coords : vec2<f32>,                    ddx : vec2<f32>, ddy : vec2<f32>                    ) -> <type>
+textureSampleGrad(texture : texture_sampled_2d<type>,         sampler : sampler, coords : vec2<f32>,                    ddx : vec2<f32>, ddy : vec2<f32>, offset : vec2<i32>) -> <type>
+textureSampleGrad(texture : texture_sampled_2d_array<type>,   sampler : sampler, coords : vec2<f32>, array_index : u32, ddx : vec2<f32>, ddy : vec2<f32>                    ) -> <type>
+textureSampleGrad(texture : texture_sampled_2d_array<type>,   sampler : sampler, coords : vec2<f32>, array_index : u32, ddx : vec2<f32>, ddy : vec2<f32>, offset : vec2<i32>) -> <type>
+textureSampleGrad(texture : texture_sampled_3d<type>,         sampler : sampler, coords : vec3<f32>,                    ddx : vec3<f32>, ddy : vec3<f32>                    ) -> <type>
+textureSampleGrad(texture : texture_sampled_3d<type>,         sampler : sampler, coords : vec3<f32>,                    ddx : vec3<f32>, ddy : vec3<f32>, offset : vec3<i32>) -> <type>
+textureSampleGrad(texture : texture_sampled_cube<type>,       sampler : sampler, coords : vec3<f32>,                    ddx : vec3<f32>, ddy : vec3<f32>                    ) -> <type>
+textureSampleGrad(texture : texture_sampled_cube_array<type>, sampler : sampler, coords : vec3<f32>, array_index : u32, ddx : vec3<f32>, ddy : vec3<f32>                    ) -> <type>
+textureSampleGrad(texture : texture_depth_2d,                 sampler : sampler, coords : vec2<f32>,                    ddx : vec2<f32>, ddy : vec2<f32>                    ) -> f32
```

Comment:
similar concern to `Bias` variants here - maybe we shouldn't have depth textures for this function at all

Reply:
Agreed. Removed.